apps/nextjs/src/components/ExportsDialogs/useExistenceCheck.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useState } from "react";
+
+import * as Sentry from "@sentry/react";
+
+export const useExistenceCheck = <T>({
+  success,
+  data,
+  chatId,
+  messageId,
+  checkFn,
+}: {
+  success?: boolean;
+  data?: T;
+  chatId: string;
+  messageId?: string;
+  checkFn: (arg: { chatId: string; data: T; messageId: string }) => void;
+}) => {
+  const [checked, setChecked] = useState(false);
+
+  const check = useCallback(() => {
+    if (success && data && !checked) {
+      try {
+        if (!messageId) {
+          throw new Error("messageId is undefined");
+        }
+        checkFn({
+          chatId,
+          data,
+          messageId,
+        });
+        setChecked(true);
+      } catch (cause) {
+        const error = new Error("Failed to check for existing export", {
+          cause,
+        });
+        Sentry.captureException(error, {
```

Comment:
Does this need some user feedback?

Reply:
Yes it should, although I haven't changed functionality here, just refactored this out to reduce duplication. Willtake a look to see if there's a simple way to add user feedback but not wanting to get pulled into widening the scope of this

Reply:
Actually just seen that this doesn't actually block the user. It's a background check, and the user hits download and would see an error if there was a subsequent error